lib/sgtn-client/api/translation.rb

```diff
@@ -19,6 +24,11 @@ def self.getString(component, key, locale)
           if str.nil?
             SgtnClient.logger.debug "[Translation][getString] Missing source string with key: #{key}, component: #{component}, locale: #{locale}"
           end
+        else
+          unless SgtnClient::LocaleUtil.is_default(locale)
+            source = SgtnClient::Source.getSource(component, key, SgtnClient::Config.configurations.default)
+            str = compareSource(component, key, SgtnClient::LocaleUtil.get_source_locale, source, str)
+          end
```

Comment:
I will have another PR to cover getStrings, it's not in this PR's scope, let's handle them separately.

Reply:
No need for source comparison in this PR then. Single-string comparison is not existing in design spec.

Reply:
the spec flow was changed too frequently, this is not good in the integration period with product, it will cause many risks.

Reply:
For the component-based comparison, it's not good to do in the API call, that will have performance issue.

Reply:
Please follow the spec and review the code suggestion.
It is not good to do source comparison at the API layer (getString and getStrings). You should have done it in the data layer, sabving correct data in cache. The spec has been updated now because the flow in Java has always been not good and not even defined in the spec.

Reply:
> For the component-based comparison, it's not good to do in the API call, that will have performance issue.

Please either follow the spec and use the code suggestion, or define the permormance issue that you have in mind.

Yes, is not good to do source comparison at the API layer (getString and getStrings), which is what you are doing here in getString. You should have done it in the data layer in the first place, and save correct data in cache. The API layer (translaiton.rb) should not have too many code logic. 

The spec has been updated now because the flow in Java client has always been incorrect, and not even defined in the spec.

Reply:
I meant component-based, not string-based, will cause the performance issue.
The spec is not safe to practice for now, we need to address the string-based comparison as high priority.